cmd/snap/export_test.go

```diff
@@ -81,6 +79,100 @@ var (
 	InterfacesDeprecationNotice = interfacesDeprecationNotice
 )
 
+func MaybePrintServices(w writeflusher, theSnap *client.Snap) {
```

Comment:
not completely a fan of these exported helpers that both build the infoWriter and call the methods.  We can keep the type opaque to tests but still have a bit more orthogonality,  for example something like, just as an example:

``` 
-func MaybePrintType(w writeflusher, theSnap *client.Snap) {
-       iw := &infoWriter{writeflusher: w, theSnap: theSnap}
+func IW(w writeflusher, theSnap *client.Snap) *infoWriter {
+       return &infoWriter{writeflusher: w, theSnap: theSnap}
+}
+
+func MaybePrintType(iw *infoWriter) {
        iw.maybePrintType()
 }
```

`IW` not the final name(!), we would have a couple of those infoWriter builders,  some could also use the setup*Snap methods.